editor/tests/org.fusesource.ide.projecttemplates.tests/src/test/java/org/fusesource/ide/projecttemplates/NameAndWeightComparatorTest.java

```diff
@@ -0,0 +1,65 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.projecttemplates;
+
+import static org.junit.Assert.assertEquals;
+
+import org.fusesource.ide.projecttemplates.wizards.pages.model.CategoryItem;
+import org.fusesource.ide.projecttemplates.wizards.pages.model.NameAndWeightComparator;
+import org.fusesource.ide.projecttemplates.wizards.pages.model.NameAndWeightSupport;
+import org.junit.Test;
+
+/**
+ * @author lhein
+ */
+public class NameAndWeightComparatorTest {
+
+	private NameAndWeightComparator comp = new NameAndWeightComparator();
+	
+	/**
+	 * tests the case where the name and the weight are equal
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testEqualNameAndWeight() throws Exception {
+		NameAndWeightSupport o1 = new CategoryItem("o1", "test", 10, null);
+		NameAndWeightSupport o2 = new CategoryItem("o2", "test", 10, null);
+		int res = comp.compare(o1, o2);
+		assertEquals(0, res);
+	}
+	
+	/**
+	 * tests the case where the weights are equal but names differ
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testEqualWeightDifferentName() throws Exception {
+		NameAndWeightSupport o1 = new CategoryItem("o1", "test1", 10, null);
+		NameAndWeightSupport o2 = new CategoryItem("o2", "test2", 10, null);
+		int res = comp.compare(o1, o2);
+		assertEquals(-1, res);
```

Comment:
I see that you added AssertJ in dependencies so you can leverage it using assertThat(res).isEqualTo(-1)
at the beginning I was not very convinced about this syntax but after using it several weeks, I find it really nicer, even more when you work with arrays or specific types (Date). For integer the interest is limited to keeping consistency with framework language in other project.
So I propose either to use ASsertJ assertions or to remove the dependency in the plugin.xml